packages/esm-appointments-app/src/home-appointments/appointments-list.component.tsx

```diff
@@ -90,17 +87,34 @@ const RenderStatus = ({ status, t, appointmentUuid, mutate }) => {
 };
 
 const AppointmentsBaseTable = () => {
-  const { useBahmniAppointmentsUI: useBahmniUI } = useConfig();
+  const { user } = useSession();
+
+  const { useBahmniAppointmentsUI: useBahmniUI, useFullViewPrivilege, fullViewPrivilege } = useConfig();
   const { isLoading, appointments } = useTodayAppointments();
 
+  const [fullView, setFullView] = useState(false);
+
+  const filteredAppointments = !fullView
+    ? appointments.filter((appointment) => appointment.status === 'Scheduled')
+    : appointments;
+
+  const userHasFullAccess = useFullViewPrivilege && userHasAccess(fullViewPrivilege, user);
```

Comment:
Whether a user has full access or not should not be dependent on if the system is currently using `useFullViewPrivilege`. Is it safe to remove the `useFullViewPrivilege` in this statement?

Reply:
From what I discussed with Romain we want to give a user the choice to decide whether they want to apply the use of privileges, that is where the boolean comes in. So He says we might not want to apply privileges for everyone so that is why he suggests we make it configurable.

So if someone doesn't want to use any privileges they will see everything, from what I understood, he meant forexample for OpenMRS we might not want to have any limitations to the table, so that's why we need that boolean.

Reply:
Yes, understood. However that logic takes place later. Here you are just defining whether they have access or not, *if* access was to be considered. Your filter logic happens next. Basically if you combine the two statements I think you would see useFullViewPrivilige twice, meaning it can be removed here

Reply:
hmmm, I think I am now confused, maybe I understood it wrongly.

Reply:
1. Your original statement is
`const userHasFullAccess = useFullViewPrivilege && userHasAccess(priv, user);
const fullView = userHasFullAccess || !useFullViewPrivilege;`

2. Substitute for userHasFullAccess
`const fullView = (useFullViewPrivilige && userHasAccess(priv, user)) || !useFullViewPrivilige`

3. Distributive law
`const fullView = (!useFullViewPrivilige || useFullViewPrivilige) && (!useFullViewPrivilige || userHasAccess(priv, user)`

4. Since `!a || a` is always true, we can eliminate it
`const fullView = !useFullViewPrivilige || userHasAccess(priv, user)`

Reply:
Yah you are right I now understand I will push a change tomorrow. Thanks